DEHPMatlab/ViewModel/DataSourceViewModel.cs

```diff
@@ -0,0 +1,121 @@
+    {
+        /// <summary>
+        /// The connect text for the connect button
+        /// </summary>
+        private const string ConnectText = "Connect";
+
+        /// <summary>
+        /// The disconnect text for the connect button
+        /// </summary>
+        private const string DisconnectText = "Disconnect";
+
+        /// <summary>
+        /// Backing field for <see cref="ConnectButtonText"/>
+        /// </summary>
+        private string connectButtonText = ConnectText;
+
+        /// <summary>
+        /// Gets the <see cref="INavigationService"/>
+        /// </summary>
+        protected readonly INavigationService NavigationService;
+
+        /// <summary>
+        /// Initializes a new <see cref="DataSourceViewModel"/>
+        /// </summary>
+        /// <param name="navigationService">The <see cref="INavigationService"/></param>
+        protected DataSourceViewModel(INavigationService navigationService)
+        {
+            this.NavigationService = navigationService;
+        }
+
+        /// <summary>
+        /// Gets or sets the name
+        /// </summary>
+        public string ConnectButtonText
+        {
+            get => this.connectButtonText;
+            set => this.RaiseAndSetIfChanged(ref this.connectButtonText, value);
+        }
+
+        /// <summary>
+        /// <see cref="ReactiveCommand{T}"/> for connecting to a data source
+        /// </summary>
+        public ReactiveCommand<object> ConnectCommand { get; set; }
+
+        /// <summary>
+        /// The <see cref="IObjectBrowserViewModel"/>
+        /// </summary>
+        public IObjectBrowserViewModel ObjectBrowser { get; set; }
+
+        /// <summary>
+        /// The <see cref="IPublicationBrowserViewModel"/>
+        /// </summary>
+        public IPublicationBrowserViewModel PublicationBrowser { get; set; }
+
+        /// <summary>
+        /// The <see cref="IHubBrowserHeaderViewModel"/>
+        /// </summary>
+        public IHubBrowserHeaderViewModel HubBrowserHeader { get; set; }
+
+        /// <summary>
+        /// Initializes the <see cref="ReactiveCommand{T}"/>
+        /// </summary>
+        protected virtual void InitializeCommands()
+        {
+            this.ConnectCommand = ReactiveCommand.Create(null, RxApp.MainThreadScheduler);
+            this.ConnectCommand.Subscribe(_ => this.ConnectCommandExecute());
+        }
+
+        /// <summary>
+        /// Executes the <see cref="ConnectCommand"/>
+        /// </summary>
+        protected abstract void ConnectCommandExecute();
+
+        /// <summary>
+        /// Updates the <see cref="ConnectButtonText"/>
+        /// </summary>
+        /// <param name="isSessionOpen">Assert whether the the button text should be <see cref="ConnectText"/> or <see cref="DisconnectText"/></param>
+        protected void UpdateConnectButtonText(bool isSessionOpen)
+        {
+            this.ConnectButtonText = isSessionOpen ? DisconnectText : ConnectText;
```

Comment:
Dont forget the `this.` qualifier on the privates.

Reply:
Those privates fields are marked as constant